lib/jit_preloader.rb

```diff
@@ -25,6 +25,14 @@ def self.globally_enabled=(value)
     @enabled = value
   end
 
+  def self.query_dive_limit=(limit)
+    @dive_limit = [limit - 1, 1].max
```

Comment:
Should we simply raise an exception if they give us a bogus query_dive_limit?

Reply:
I am more leaning to either fall back to not having dive limit if we want to change that.
If it raise exception, it will mean all preload process in manage(when we apply the newer gem) will fail which is a much more destructive outcome IMO.